src/jinja2/parser.py

```diff
@@ -641,14 +638,19 @@ def parse_unary(self, with_filter: bool = True) -> nodes.Expr:
             node = self.parse_filter_expr(node)
         return node
 
-    def parse_primary(self) -> nodes.Expr:
+    def parse_primary(self, with_namespace: bool = False) -> nodes.Expr:
         token = self.stream.current
         node: nodes.Expr
         if token.type == "name":
             if token.value in ("true", "false", "True", "False"):
                 node = nodes.Const(token.value in ("true", "True"), lineno=token.lineno)
             elif token.value in ("none", "None"):
                 node = nodes.Const(None, lineno=token.lineno)
+            elif with_namespace and self.stream.look().type == "dot":
+                next(self.stream)  # token
```

Comment:
Should the order of these comments be reversed? It seems like they're backwards compared to the removed previous code above.